pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -352,3 +377,71 @@ func isMaster(machine mapiv1.Machine, client client.Client) bool {
 	}
 	return false
 }
+
+// isMachineDisruptionAllowed returns true if the provided MachineDisruptionBudget allows any disruption
+func isMachineDisruptionAllowed(mdb *healthcheckingv1alpha1.MachineDisruptionBudget, maxDisruptedMachinSize int) bool {
+	if mdb.Status.ObservedGeneration < mdb.Generation {
+		glog.Warningf("The machine disruption budget %s is still being processed by the server", mdb.Name)
+		return false
+	}
+	if mdb.Status.MachineDisruptionsAllowed < 0 {
+		glog.Warningf("The machine disruption budget %s MachineDisruptionsAllowed is negative", mdb.Name)
+		return false
+	}
+	if len(mdb.Status.DisruptedMachines) > maxDisruptedMachinSize {
+		glog.Warningf("The machine disruption budget %s DisruptedMachines map too big - too many deletions not confirmed by MDB controller", mdb.Name)
+		return false
+	}
+	if mdb.Status.MachineDisruptionsAllowed == 0 {
+		glog.Warningf("Cannot remediate machine as it would violate the machine's disruption budget %s", mdb.Name)
+		return false
+	}
+
+	return true
+}
+
+func (r *ReconcileMachineHealthCheck) decrementMachineDisruptionsAllowed(machineName string, mdb *healthcheckingv1alpha1.MachineDisruptionBudget) error {
+	mdb.Status.MachineDisruptionsAllowed--
+	if mdb.Status.DisruptedMachines == nil {
+		mdb.Status.DisruptedMachines = make(map[string]metav1.Time)
+	}
+	// MachineHealthCheck controller needs to inform the MDB controller that it is about to remediate a machine
+	// so it should not consider it as available in calculations when updating MachineDisruptions allowed.
+	// If the machine is not remediated within a reasonable time limit MDB controller will assume that it won't
+	// be remediated at all and remove it from DisruptedMachines map.
+	mdb.Status.DisruptedMachines[machineName] = metav1.Time{Time: time.Now()}
+	return r.client.Update(context.TODO(), mdb)
+}
+
+// retryCheckAndDecrement tries to run checkAndDecrement command and if it fails because of the
```

Comment:
Can you please move this to the MDB controller
Also what is command? what is check? what is decrementing? Naming is confusing here

Reply:
Updated the name and the comment and moved under MDB controller

Reply:
done